crates/flox-rust-sdk/src/models/publish.rs

```diff
@@ -1196,4 +1224,14 @@ mod tests {
 
         assert_eq!(snapshot_actual, snapshot_actual);
     }
+
+    #[tokio::test]
+    async fn resolves_indirect_ref_to_gitfile() {
+        let indirect_ref = FlakeRef::from_url("flake:flox", PARSER_UTIL_BIN_PATH).unwrap();
+        let (flox, _temp_dir_handle) = flox_instance();
+        let publishable = PublishFlakeRef::from_flake_ref(indirect_ref, &flox, true)
+            .await
+            .unwrap();
+        assert!(matches!(publishable, PublishFlakeRef::Https(_)));
```

Comment:
Optional (just a bit easier for me to read a test when the data is concrete):
```suggestion
        assert_eq!(
            publishable,
            PublishFlakeRef::Https(
                GitRef::from_str("git+https://github.com/flox/floxpkgs?ref=master").unwrap()
            )
```